src/main/python/run_es_regression.py

```diff
@@ -0,0 +1,235 @@
+                if output:
+                    logger.info('subprocess - ' + output.decode('utf8').strip())
+                    if capture: arr.append(output.decode('utf8').strip())
+            # Make sure we read the output completely
+            while True:
+                output = process.stdout.readline()
+                if not output: break
+                logger.info('subprocess - ' + output.decode('utf8').strip())
+                if capture: arr.append(output.decode('utf8').strip())
+            if capture: return arr
+            else: return process.returncode
+
+    def is_alive(self):
+        try:
+            response = requests.get('http://localhost:9200/')
+            response.raise_for_status()
+        except: return False
+        else: return True
+
+    def does_index_exist(self, collection):
+        # Make sure ES is alive:
+        if self.is_alive():
+            try:
+                response = requests.get('http://localhost:9200/{}'.format(collection))
+                response.raise_for_status()
+            except: return False
+            else: return True
+        else: raise Exception('ES does not appear to be alive!')
+
+    def delete_index(self, collection):
+        logger.info('Deleting index {}...'.format(collection))
+        # Make sure the index exists:
+        if self.does_index_exist(collection):
+            try:
+                response = requests.request('DELETE', url='http://localhost:9200/{}'.format(collection))
+                response.raise_for_status()
+            except: return False
+            else: return True
+        else: raise Exception('The index {} does not exist!'.format(collection))
+
+    def create_index(self, collection):
+        logger.info('Creating index {}...'.format(collection))
+        # Make sure the index does not exist:
+        if not self.does_index_exist(collection):
+            filename = 'src/main/resources/elasticsearch/index-config.{}.json'.format(collection)
+            if not os.path.exists(filename):
+                raise Exception('No config found in src/main/resources/elasticsearch/ for {}!'.format(collection))
+            logger.info('Using index config for {} at {}'.format(collection, filename))
+            with open(filename, mode='r') as file:
+                json = file.read()
+            try:
+                response = requests.request('PUT', url='http://localhost:9200/{}'.format(collection),
+                                            data=json, headers={'Content-type': 'application/json'})
+                response.raise_for_status()
+            except:
+                logger.info(response)
+                return False
+            else:
+                return True
+        else:
+            raise Exception('The index {} already exists!'.format(collection))
+
+    def insert_docs(self, collection, path):
+        logger.info('Inserting documents from {} into {}... '.format(path, collection))
```

Comment:
One more small thing: do we want to check if `path` here is valid? And throw an exception if not.
For example, when I tried running `python src/main/python/run_es_regression.py --regression msmarco-passage --input random_non_existed_path`, the program throw an exception from java, but the python program resumes: sleeping for 60s, do evaluation, and finally give me a runtime error..

Reply:
Good catch. Will add.